### PR TITLE
Add conversion of GEMM ONNX to TOSA

### PR DESCRIPTION
**Pull request by maxbartel**

### CODE REVIEW

File: src/Conversion/ONNXToTOSA/Math/Gemm.cpp [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+public:
+  using OpConversionPattern<ONNXGemmOp>::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(ONNXGemmOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    TosaBuilder tosaBuilder(rewriter, op->getLoc());
+    // If legal, create a FullyConnected operator instead
+    if (rewriteToTosaFC(op, adaptor, rewriter, tosaBuilder)) {
+      return success();
+    }
```
Comment (ljfitz) [inline src/Conversion/ONNXToTOSA/Math/Gemm.cpp:37]: Forgotten if I've nit-picked on this before - does onnx-mlir not follow https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements?

Comment (AlexandreEichenberger) [reply src/Conversion/ONNXToTOSA/Math/Gemm.cpp:37]: We recommend it.

File: src/Conversion/ONNXToTOSA/Math/Gemm.cpp [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+} // namespace
+
+void populateLoweringONNXGemmOpToTOSAPattern(ConversionTarget &target,
+    RewritePatternSet &patterns, TypeConverter &typeConverter,
+    MLIRContext *ctx) {
+  patterns.insert<ONNXGemmOpLoweringToTOSA>(typeConverter, ctx);
+}
+
+} // namespace onnx_mlir
```
Comment (AlexandreEichenberger) [inline src/Conversion/ONNXToTOSA/Math/Gemm.cpp:244]: nit: we typically add an empty line at the end of the file.

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM pending the small formatting fixes

**Review by eric-k256 — APPROVED** [review]
From a TOSA perspective, this looks good to me.

File: src/Conversion/ONNXToTOSA/Math/Gemm.cpp (hunk 1 above)
Comment (maxbartel) [reply src/Conversion/ONNXToTOSA/Math/Gemm.cpp]: I removed the braces according to the LLVM Coding Standards.

**Comment by maxbartel** [comment]
@AlexandreEichenberger Could you merge this please if all pipelines are green?

**Comment by AlexandreEichenberger** [comment]
done, thanks for the contribution

**Pull request merged**